CMakeLists.txt

```diff
@@ -1,131 +1,161 @@
+option(ENABLE_SOLV_URPMREORDER "Build with support for URPM-like solution reordering?" OFF)
 
-ADD_DEFINITIONS(-DGETTEXT_DOMAIN=\\"libdnf\\")
-ADD_DEFINITIONS(-DG_LOG_DOMAIN=\\"libdnf\\")
 
-INCLUDE_DIRECTORIES (${CMAKE_SOURCE_DIR} libdnf/utils/)
+# load pkg-config first; it's required by other modules
+find_package(PkgConfig REQUIRED)
+if(APPLE)
+    set(ENV{PKG_CONFIG_PATH} "$ENV{PKG_CONFIG_PATH}:/usr/local/lib64/pkgconfig")
+    set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH};/usr/local/share/cmake/Modules/)
+    include_directories(/usr/local/include)
+endif()
 
-OPTION(WITH_GTKDOC "Enables libdnf GTK-Doc HTML documentation" ON)
-OPTION(WITH_MAN "Enables hawkey man page generation" ON)
-OPTION(WITH_HTML "Enables hawkey HTML generation" ON)
 
-OPTION(WITH_BINDINGS "Enables python/SWIG bindings" ON)
+# build dependencies
+find_package(Gpgme REQUIRED)
+find_package(LibSolv 0.6.30 REQUIRED COMPONENTS ext)
+find_package(OpenSSL REQUIRED)
 
-OPTION (ENABLE_SOLV_URPMREORDER "Build with support for URPM-like solution reordering?" OFF)
-option (ENABLE_RHSM_SUPPORT "Build with Red Hat Subscription Manager support?" OFF)
 
-# hawkey dependencies
-find_package (PkgConfig REQUIRED)
-SET (CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/modules)
-IF (${CMAKE_VERSION} VERSION_LESS 3)
-    SET (CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH};${CMAKE_SOURCE_DIR}/cmake/modules-cmake-2)
-ENDIF ()
-if (APPLE)
-    SET(ENV{PKG_CONFIG_PATH} "$ENV{PKG_CONFIG_PATH}:/usr/local/lib64/pkgconfig")
-    SET(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH};/usr/local/share/cmake/Modules/)
-endif(APPLE)
-PKG_CHECK_MODULES(GLIB gio-unix-2.0>=2.46.0 REQUIRED)
-pkg_check_modules(SMARTCOLS REQUIRED smartcols)
-FIND_LIBRARY (RPMDB_LIBRARY NAMES rpmdb)
-find_package (LibSolv 0.6.30 REQUIRED COMPONENTS ext)
-if (ENABLE_RHSM_SUPPORT)
-    pkg_check_modules (RHSM REQUIRED librhsm>=0.0.3)
-    include_directories (${RHSM_INCLUDE_DIRS})
-    add_definitions (-DRHSM_SUPPORT)
-endif ()
-pkg_check_modules (CHECK REQUIRED check)
-pkg_check_modules (REPO REQUIRED librepo)
-pkg_check_modules (LIBMODULEMD REQUIRED modulemd)
-FIND_PACKAGE(Gpgme REQUIRED)
-FIND_PACKAGE (OpenSSL REQUIRED)
-
-FIND_PROGRAM (VALGRIND_PROGRAM NAMES valgrind PATH /usr/bin /usr/local/bin)
-
-IF (ENABLE_SOLV_URPMREORDER)
-   ADD_DEFINITIONS(-DLIBSOLV_FLAG_URPMREORDER=1)
-ENDIF()
-
-if (APPLE)
-    INCLUDE_DIRECTORIES(/usr/local/include)
-endif(APPLE)
-INCLUDE_DIRECTORIES(${REPO_INCLUDE_DIRS})
-
-INCLUDE_DIRECTORIES(${GLIB_INCLUDE_DIRS})
+# build dependencies via pkg-config
+pkg_check_modules(CHECK REQUIRED check)
+pkg_check_modules(GLIB gio-unix-2.0>=2.46.0 REQUIRED)
+include_directories(${GLIB_INCLUDE_DIRS})
+pkg_check_modules(JSONC json-c)
+include_directories(${JSONC_INCLUDE_DIRS})
+pkg_check_modules(LIBMODULEMD REQUIRED modulemd)
+pkg_check_modules(REPO REQUIRED librepo)
+include_directories(${REPO_INCLUDE_DIRS})
 link_directories(${REPO_LIBRARY_DIRS})
+pkg_check_modules(RPM REQUIRED rpm)
```

Comment:
I'm pretty sure we have a minimum version that we work with, right?

Reply:
I've changed that based on minimal version from the spec file.